### PR TITLE
doctl: Add support for custom domains in Spaces CDN

### PR DESCRIPTION
**Pull request by xornivore**
- Update `cdn` command to provide support for configuring custom domains
- Update `godo` version to `v1.7.5`
- Update mocks
- Automatically set `IsSetMap` on `TestConfig.Set`
- Simplify tests in `./commands/projects_tests.go` which were previously setting `IsSetMap`

### CODE REVIEW

**Comment by mikejholly** [comment]
LGTM

**Comment by mjsteger** [comment]
LGTM

File: commands/projects_test.go [hunk 1]
```diff
@@ -96,14 +96,6 @@ func TestProjectsUpdateAllAttributes(t *testing.T) {
 		}
 		tm.projects.On("Update", projectUUID, updateReq).Return(&testProject, nil)
 
-		config.Doit.(*TestConfig).IsSetMap = map[string]bool{
```
Comment (hilary) [inline commands/projects_test.go]: How do these removals relate to adding support for custom domains in Spaces CDN?

Comment (xornivore) [reply commands/projects_test.go]: I updated `TestConfig.Set` to add entries in `IsSetMap` automatically so these are redundant now. One of the tests in this file had arguments supplied via `Set` out of sync with mutation of `IsSetMap` here: https://github.com/digitalocean/doctl/pull/415/files/789465cc23fd6b4c7c4f1683f985e55cb8d3cd3b#diff-0f9818a0fd8799f01ac7bd4c2ba02fedL148

Comment (hilary) [reply commands/projects_test.go]: Gotcha. Thanks for the explanation.

**Pull request merged**